bundles/org.openhab.binding.gpio/src/main/java/org/openhab/binding/gpio/internal/handler/PigpioDigitalOutputHandler.java

```diff
@@ -52,34 +63,182 @@ public class PigpioDigitalOutputHandler implements ChannelHandler {
      * @param updateStatus Is called when the state should be changed
      * 
      * @throws PigpioException Can be thrown by Pigpio
-     * @throws NoGpioIdException Is thrown when no gpioId is defined
+     * @throws ChannelConfigurationException Thrown on configuration error
      */
-    public PigpioDigitalOutputHandler(GPIOOutputConfiguration configuration, JPigpio jPigpio,
-            Consumer<State> updateStatus) throws PigpioException, NoGpioIdException {
+    public PigpioDigitalOutputHandler(GPIOOutputConfiguration configuration, ScheduledExecutorService scheduler,
+            Consumer<State> updateStatus) throws PigpioException, ChannelConfigurationException {
         this.configuration = configuration;
+        this.gpioId = configuration.gpioId;
+        this.scheduler = scheduler;
         this.updateStatus = updateStatus;
-        Integer gpioId = configuration.gpioId;
-        if (gpioId == null) {
-            throw new NoGpioIdException();
+
+        if (this.gpioId <= 0) {
+            throw new ChannelConfigurationException("Invalid gpioId value.");
+        }
+
+        if (configuration.pulse.compareTo(BigDecimal.ZERO) > 0) {
+            try {
+                this.pulseTimeout = configuration.pulse.intValue();
+            } catch (Exception e) {
+                throw new ChannelConfigurationException("Invalid expire value.");
+            }
+        }
+
+        if (configuration.pulseCommand.length() > 0) {
+            this.pulseCommand = configuration.pulseCommand.toUpperCase();
+            if (!"ON".equals(pulseCommand) && !"OFF".equals(pulseCommand) && !"BLINK".equals(pulseCommand)) {
```

Comment:
I would suggest to move this strings into the binding co0nstants file. they are used in more places